code/game/machinery/Sleeper.dm

```diff
@@ -229,10 +288,30 @@
 	return
 
 
-/obj/machinery/sleeper/attackby(obj/item/weapon/grab/G as obj, mob/user as mob)
-	if((!( istype(G, /obj/item/weapon/grab)) || !( ismob(G.affecting))))
+/obj/machinery/sleeper/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(iscrowbar(W) && occupant)

```

Comment:
That format is used, but it's lower down. What you're seeing is an escape case that prevents you from deconstructing the machine while someone is inside.